ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py

```diff
@@ -316,6 +332,71 @@ def _get_port_channel_member_cfg(self, p2p_link: dict, member: dict) -> dict:
             },
         }
 
+    # l3_interfaces here
     @cached_property
-    def _p2p_links_sflow(self) -> bool | None:
+    def _l3_interfaces_profiles(self) -> list:
+        return get(self._hostvars, f"{self.data_model}.l3_interfaces_profiles", default=[])
+
+    @cached_property
+    def _l3_interfaces(self) -> list:
+        return get(self._hostvars, f"{self.data_model}.l3_interfaces", default=[])
+
+    @cached_property
+    def _l3_interfaces_sflow(self) -> bool | None:
         return get(self._hostvars, f"fabric_sflow.{self.data_model}")
+
+    @cached_property
+    def _filtered_l3_interfaces(self) -> list:
+        """
+        Returns a filtered list of l3_interfaces, which only contains interfaces with our hostname.
+        For each interface any referenced profiles are applied.
+        """
+        if not (l3_interfaces := self._l3_interfaces):
+            return []
+
+        # Apply l3_interfaces._profile if set. Silently ignoring missing profile.
+        if self._l3_interfaces_profiles:
+            l3_interfaces = [self._apply_profile("l3_interfaces", l3_interface) for l3_interface in l3_interfaces]
+
+        # Filter to only include l3_interfaces with our hostname as node
+        l3_interfaces = [l3_interface for l3_interface in l3_interfaces if self.shared_utils.hostname == get(l3_interface, "node", required=True)]
+        if not l3_interfaces:
+            return []
+
+        return l3_interfaces
+
+    def _get_l3_interface_cfg(self, l3_interface: dict) -> dict | None:
+        """
+        Returns structured_configuration for one L3 Physical interface
+        """
+        if l3_interface["node"] != self.shared_utils.hostname:
+            return None
+
+        interface_name = l3_interface["interface"]
+
+        # TODO move this to description module?
+        interface_description = l3_interface.get("description") or " ".join(
+            [elem for elem in [l3_interface.get("peer"), l3_interface.get("peer_interface")] if elem is not None]
+        )
```

Comment:
You should have a final `or None` so you don't get an empty string as description (will error on EOS)